rai/node/node.cpp

```diff
@@ -1527,6 +1527,17 @@ rai::node::~node ()
 	stop();
 }
 
+void rai::node::copy_with_compaction ()
+{
+	auto vacuum_path = application_path / "vacuumed.ldb";
```

Comment:
The output file path should probably an argument.

This might not need to be a function either, since it's pretty common practice throughout the code to access node.store directly.

Reply:
Not sure why that would be useful. The fact that it's a separate file is an implementation detail that the user shouldn't are about. In fact, it should probably write to a tmp-file with a unique name.

Reply:
How does the caller know what the output file name is?

Reply:
Well, that would need to be communicated if I change this to use tmp-files. But for now, using a fixed name seems reasonable enough - it's not like anyone should put conflicting ldb files into the data folder :)

Reply:
You could move this entirely into the other block, since it only has one caller, and only one potential caller with the hardcoded name - and avoid spreading the hardcoded names any further.

Reply:
What do you mean by other block?

Reply:
```
auto data_path = application_path / "data.ldb";
auto vacuum_path = application_path / "vacuumed.ldb";

bool success = false;
{
    inactive_node node (data_path);
    success = !mdb_env_copy2 (node.node->store.environment.environment, vacuum_path.string ().c_str (), MDB_CP_COMPACT);
}

if (success)
{
   std::cout << "Finalizing" << std::endl;
   // Note that these throw on failure
   boost::filesystem::remove (data_path);
   boost::filesystem::rename (vacuum_path, data_path);
   std::cout << "Vacuum completed" << std::endl;
}
```

Reply:
Or, if it's really a useful function that needs to be on node, it should be more generic and either return the path or take a path. Either way makes sense.

Reply:
Yeah, I can make it take a path, makes sense.